### PR TITLE
Minor

### PR DESCRIPTION
**Pull request by CaseyCarter**
Several small, independent changes not worth individual pull requests.

### CODE REVIEW

File: include/range/v3/detail/optional.hpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ namespace ranges
             optional(T t)
               : data_(emplaced_index<1>, std::move(t))
             {}
-            template <typename...Args, CONCEPT_REQUIRES_(Constructible<T, Args &&...>())>
+            template <typename...Args, CONCEPT_REQUIRES_(Constructible<T, Args...>())>
```
Comment (ericniebler) [inline include/range/v3/detail/optional.hpp]: Why?

Comment (CaseyCarter) [inline include/range/v3/detail/optional.hpp]: Equivalent but shorter.

Comment (ericniebler) [inline include/range/v3/detail/optional.hpp]: Are you not forgetting that `Concept<T>()` in range-v3 means `Concept<T&>()`?

Comment (CaseyCarter) [inline include/range/v3/detail/optional.hpp]: Not this time ;) `Constructible` have been cured of that madness and use `std::declval`. I REALLY need to find some time to transition the rest of the library.

Comment (ericniebler) [inline include/range/v3/detail/optional.hpp]: >  I REALLY need to find some time to transition the rest of the library

I would love such a change, but until then, I prefer the consistency. Please stick to T&& for now, as annoying as it is.

Comment (CaseyCarter) [inline include/range/v3/detail/optional.hpp]: Fair enough.

**Comment by CaseyCarter** [comment]
Squashed without the `optional` change.

**Pull request merged**